meson.build

```diff
@@ -4,7 +4,7 @@ project('hardware_isolation', 'cpp', 'c',
     default_options: [
       'buildtype=debugoptimized',
       'warning_level=3',
-      'cpp_std=c++20',
+      'cpp_std=c++23',
       'werror=true',
     ],
     meson_version: '>=0.58.0',
```

Comment:
You need to upgrade meson version as well `1.1.1` like this https://github.com/ibm-openbmc/phosphor-logging/pull/67